### PR TITLE
Publish weco-deploy images to ECR Public; simplify the build process

### PR DESCRIPTION
**Pull request by alexwlchan**
These changes are with an eye to https://github.com/wellcomecollection/platform/issues/5545; I imagine I’ll need to do some tweaking to get them working on main. I’ve already fixed the issue with publishing to PyPI (which was an incorrect Content-Type; see https://github.com/wellcomecollection/weco-deploy/commit/1a75b500631a9b04b9cacef936ca74b3a21cd1da).

Changes:

* Define the ECR repository in Terraform (and tag it so we know where it's defined)
* Add a new ECR Public repository for weco-deploy
* Replace a somewhat unwieldy Python build script that shells out to a bunch of other tools with a simpler shell script
* Publish images to Docker Hub, ECR and ECR Public
* Use an ECR public image for the weco-deploy image base, not our vendored version

Strictly speaking we don't need ECR Public here, but it makes onboarding easier if all our tooling pulls from public repositories, and it makes it easier to find places we're still using ECR Private in the build system.

### CODE REVIEW

File: scripts/publish_docker_images.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+done
+
+eval $(aws ecr get-login --no-include-email)
+for version in $VERSION_TAGS
+do
+  docker tag "$LATEST_TAG" "760097843905.dkr.ecr.eu-west-1.amazonaws.com/$LATEST_TAG"
+  docker push "760097843905.dkr.ecr.eu-west-1.amazonaws.com/$LATEST_TAG"
+done
+
+aws ecr-public get-login-password --region us-east-1 | xargs -I '{}' docker login --username AWS --password '{}'
```
Comment (jamieparkinson) [inline scripts/publish_docker_images.sh:61]: I'm not sure that I agree that this script is any simpler than the Python one (see also `tr '.' ' ' | awk '{print $3}'`), it seems fairly equivalent to me, about the same number of lines modulo comments...

**Review by jamieparkinson — APPROVED** [review]

File: scripts/publish_docker_images.sh (hunk 1 above)
Comment (alexwlchan) [reply scripts/publish_docker_images.sh:61]: The line count may be similar, but the Python implementation is unnecessarily invoked via tox, which has its own Dockerfile, and the code for which is scattered around half a dozen files.

Comment (jamieparkinson) [reply scripts/publish_docker_images.sh:61]: We're still using tox for pypi publishing though right?

Comment (alexwlchan) [reply scripts/publish_docker_images.sh:61]: Yes, but not for this aspect of the build. I would like to de-tox this repo; I don't think we get much benefit from it.

**Comment by alexwlchan** [comment]
Closing in lieu of #120.

**Pull request closed without merging**